pipeline/beam_tables.py

```diff
@@ -289,9 +360,13 @@ def _data_to_load(self,
         'gs://firehook-scans/echo/CP_Quack-echo-2020-08-23-06-01-02/results.json']
     """
     if incremental_load:
-      full_table_name = self._get_full_table_name(table_name)
-      existing_sources = _get_existing_datasources(full_table_name,
-                                                   self.project)
+      if table_name:
+        full_table_name = self._get_full_table_name(table_name)
+        existing_sources = _get_existing_bq_datasources(full_table_name,
+                                                        self.project)
+      elif gcs_folder:
```

Comment:
add an else clause that raises an error if both are None.

Reply:
done